### ISSUE TITLE
Can't consume a C# module in React Native 0.62

### ISSUE THREAD

**Issue by qmatteoq** [microsoft/react-native-windows#6904]
## Environment
Run the following in your terminal and copy the results here.
1. `npx react-native --version`: 4.13.1
2. `npx react-native info`: 

```
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 4.89 GB / 15.60 GB
  Binaries:
    Node: 12.9.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.10.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.14393.0, 10.0.15063.0, 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.8.30907.101 (Visual Studio Enterprise 2019), 16.9.30803.129 (Visual Studio Enterprise 2019)
  Languages:
    Java: Not Found
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0
    react-native: ^0.62.0-0 => 0.62.2
    react-native-windows: ^0.62.0-0 => 0.62.21
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Create a new project based on React Native 0.62 and add Windows support
2. Open package.json and, under dependencies, add the following entry:

```
 "react-native-view-shot": "qmatteoq/react-native-view-shot#v0.62"
```
This is a 3rd party module built with C# and targeting 0.62.

3. Manually link the module, by adding the following entry in the pch.h file:

```
#include <winrt/RNViewShot.h>
```

4. Move to the App.cpp file and, before `InitializeComponent()`, add the following entry:

```
PackageProviders().Append(winrt::RNViewShot::ReactPackageProvider());
```

5. Build and run the project.

## Expected Results
The application starts and I can use the native module. Instead, the application crashes at startup.
Based on the information provided [in the troubleshooting page](https://microsoft.github.io/react-native-windows/docs/native-modules-troubleshooting), this is a known issue and you must add the following entry in the C++ project's file to fix it:

```
<ConsumeCSharpModules>true</ConsumeCSharpModules>
```

However, if I do this, the solution stops building and Visual Studio returns the following error:

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\NuGet\16.0\Microsoft.NuGet.targets(389,5): error : Your project does not reference ".NETCore,Version=v5.0" framework. Add a reference to ".NETCore,Version=v5.0" in the "frameworks" section of your project.json and then re-run NuGet restore.
```

**Comment by asklar** [comment]
Update - different versions of RNW and VS (via the uwp runtime versions) behave differently:

| | UWP 2.2.10 | UWP 2.2.11 |
| 0.62 | ❌ | ❌ |
| 0.63 | ✅ | ❌ |

I'm fixing the combination of RNW 0.63 + UWP 2.2.11 (VS 16.9) in #6925 ; @qmatteoq is going to talk to the customer to upgrade to 0.63

**Comment by chrisglein** [comment]
Won't be fixed in 0.62. Will be fixed in 0.63 including the asterisk of #6925. Hopefully that works for them.
